package/system/zram-swap/files/zram.init

```diff
@@ -163,16 +163,15 @@ start()
 	local zram_size="$( zram_getsize )"
 	local zram_dev="$( zram_getdev )"
 	zram_applicable "$zram_dev" || return 1
-	local zram_priority="$( uci -q get system.@system[0].zram_priority )"
-	zram_priority=${zram_priority:+-p $zram_priority}
+	local zram_priority=`cat /proc/swaps | tail -n +2 | awk '{print $5}' | sort | tail -n 1 | { grep . || echo -1; }`
```

Comment:
Please wrap lines with more than 74 chars.
Using backticks is discouraged.
cat + tail can probably be replaced by just tail?
What's grep . || echo -1 good for?

Reply:
> Please wrap lines with more than 74 chars.
> Using backticks is discouraged.
> What's grep . || echo -1 good for?
if the value is empty, it will return -1 , and it only works in backticks, I not sure if I could reduce it to < 74 chars

> cat + tail can probably be replaced by just tail?
Thanks, I replaced it with tail

Reply:
> > Using backticks is discouraged.
> > What's grep . || echo -1 good for?
> 
> if the value is empty, it will return -1 , and it only works in backticks, I not sure if I could reduce it to < 74 chars

No. Which shell are you testing this with? If I do
`root@OpenWrt:~# echo $(tail -n +2 /proc/swaps | awk '{print $5}' | sort | tail -n 1 | { grep . || echo -1; })`
this works whether I have swap devices set up (returning the highest priority) or or (returning -1).

In addition, your sorting is wrong because you don't use numeric sort. Try this:
`root@OpenWrt:~# printf "-1\n50\n1\n100\n" | sort`
`-1`
`1`
`100`
`50`

And then try with numeric sort:
`root@OpenWrt:~# printf "-1\n50\n1\n100\n" | sort -n`
`-1`
`1`
`50`
`100`

Reply:
> > cat + tail can probably be replaced by just tail?
> 
> Thanks, I replaced it with tail

You can even replace tail + awk with just awk here:
`awk 'NR>1 {print $5}' /proc/swaps`

Reply:
> In addition, your sorting is wrong because you don't use numeric sort. Try this:
> `root@OpenWrt:~# printf "-1\n50\n1\n100\n" | sort`
> `-1`
> `1`
> `100`
> `50`
> 
> And then try with numeric sort:
> `root@OpenWrt:~# printf "-1\n50\n1\n100\n" | sort -n`
> `-1`
> `1`
> `50`
> `100`
Thanks, I did not notice that

> No. Which shell are you testing this with? If I do
> `root@OpenWrt:~# echo $(tail -n +2 /proc/swaps | awk '{print $5}' | sort | tail -n 1 | { grep . || echo -1; })`
> this works regardless of whether I have swap devices set up (returning the highest priority) or not (returning -1).
Well, it doesn't work when I tried last time, but yes it seems to be working now, thanks

Reply:
> > > cat + tail can probably be replaced by just tail?
> > 
> > 
> > Thanks, I replaced it with tail
> 
> You can even replace tail + awk with just awk here:
> `awk 'NR>1 {print $5}' /proc/swaps`

Oh, and since I happen to like awk, you don't even need your awk + sort + tail + grep || magic altogether. Here's an awk oneliner:
`awk 'BEGIN {p=-1}; NR>1 {if ($5 >= p) {p=$5 + 1}}; END {print p}' /proc/swaps`

Reply:
> You can even replace tail + awk with just awk here:
> `awk 'NR>1 {print $5}' /proc/swaps`

Thanks, replaced

Reply:
> and since I happen to like awk, you don't even need your awk + sort + tail + grep || magic altogether. Here's an awk oneliner:
> `awk 'BEGIN {p=-1}; NR>1 {if ($5 >= p) {p=$5 + 1}}; END {print p}' /proc/swaps`

That line got much smaller, than what I actually wrote at first, Thanks